ddtrace/internal/sampling.py

```diff
@@ -206,30 +218,22 @@ def apply_span_sampling_tags(self, span):
         span.set_metric(_SINGLE_SPAN_SAMPLING_MECHANISM, SamplingMechanism.SPAN_SAMPLING_RULE)
         span.set_metric(_SINGLE_SPAN_SAMPLING_RATE, self._sample_rate)
         # Only set this tag if it's not the default -1
-        if self._max_per_second != -1:
+        if self._max_per_second != _SINGLE_SPAN_SAMPLING_MAX_PER_SEC_NO_LIMIT:
             span.set_metric(_SINGLE_SPAN_SAMPLING_MAX_PER_SEC, self._max_per_second)
 
 
 def get_span_sampling_rules():
     # type: () -> List[SpanSamplingRule]
     json_rules = _get_span_sampling_json()
+    validate(json_rules, SPAN_SAMPLING_JSON_SCHEMA)
```

Comment:
do we have an idea of the performance impact of this validation?

Reply:
Not currently, I can do that as follow up work as discussed internally.